DeepSpeech.py

```diff
@@ -1170,17 +1174,17 @@ def _next_epoch(self):
         if (FLAGS.early_stop is True) and (FLAGS.validation_step > 0) and (len(self._dev_losses) >= FLAGS.earlystop_nsteps):
 
             # Calculate the mean of losses for past epochs
-            mean_loss = np.mean(self._dev_losses[-FLAGS.earlystop_nsteps:-2])
+            mean_loss = np.mean(self._dev_losses[-FLAGS.earlystop_nsteps:-1])
             # Calculate the standard deviation for losses from validation part in the past epochs
-            std_loss = np.std(self._dev_losses[-FLAGS.earlystop_nsteps:-2])
+            std_loss = np.std(self._dev_losses[-FLAGS.earlystop_nsteps:-1])
             # Update the list of losses incurred
             self._dev_losses = self._dev_losses[-FLAGS.earlystop_nsteps:]
-            log_debug('Current validation loss: %f, std_of_loss(n_step) :%f mean_loss(n_step): %f' % (self._dev_losses[-1], std_loss, mean_loss))
+            log_debug('Checking for early stopping (last %d steps) validation loss: %f, with standard deviation: %f and mean: %f' % (FLAGS.earlystop_nsteps, self._dev_losses[-1], std_loss, mean_loss))
 
             # Making sure slight fluctuations don't bother the early stopping from performing
```

Comment:
Maybe expand this comment clarifying the two different things we're doing here? Something like:

```python
            # Check if the validation loss has started increasing, or is not
            # decreasing substantially, making sure slight fluctuations don't
            # bother the early stopping from working.
```